Logic/bot.py

```diff
@@ -25,12 +25,15 @@ def find_holes(self, frame):
         '''Responsible for finding the holes if not set'''
 
         if not self.holes:
-            self.holes = self.ball_detection.find_holes(frame)
+            corner_holes = self.ball_detection.find_corner_holes(frame)
 
-            board_positions = self.ball_detection.board_boundary(self.holes)
+            if len(corner_holes) == 4:
```

Comment:
How does this know that the hole position is correct? Or does the corner hole detection have strict rules and only finds the 4 corners?

Reply:
This does not ensure that the 4 detected holes are correct, however, when applying the HoughCircles it uses the minimum and maximum radius of the corner holes.